eth/eventservices/roundservice.go

```diff
@@ -70,38 +70,62 @@ func (s *RoundsService) Stop() error {
 func (s *RoundsService) tryInitializeRound(blkNum *big.Int, blkHash common.Hash) (bool, error) {
 	currentRound, err := s.client.CurrentRound()
 	if err != nil {
-		return false, err
+		return true, err
 	}
 
 	lastInitializedRound, err := s.client.LastInitializedRound()
 	if err != nil {
-		return false, err
+		return true, err
 	}
 
 	if lastInitializedRound.Cmp(currentRound) == -1 {
 		roundLength, err := s.client.RoundLength()
 		if err != nil {
-			return false, err
+			return true, err
 		}
 
 		currentRoundStartBlock := new(big.Int).Mul(currentRound, roundLength)
 
 		shouldInitialize, err := s.shouldInitializeRound(currentRoundStartBlock, blkNum, blkHash)
 		if err != nil {
-			return false, err
+			return true, err
 		}
 
 		if shouldInitialize {
 			glog.Infof("New round - preparing to initialize round to join active set")
 
 			tx, err := s.client.InitializeRound()
 			if err != nil {
-				return false, err
+				return true, err
 			}
 
 			err = s.client.CheckTx(tx)
 			if err != nil {
-				return false, err
+				txErr := err
```

Comment:
hmm... Shouldn't we check whether the round was initialized **before** calling `InitializeRound`?

Reply:
So we do check whether the user should call `InitializeRound` before actually submitting the transaction - it consists of two conditions: 1) the current round is not initialized on L81 and 2) the user will join the upcoming active transcoder set in the round and has been randomly selected to call `InitializeRound` on L94.

This additional check of whether the current round is initialized **after** calling `InitializeRound` is to determine whether a race condition was triggered - it is possible that the user was randomly selected to call `InitializeRound`, submitted the transaction and before the transaction was mined, another user manually submits a `InitializeRound` transaction that is mined first. As a result, the original `InitializeRound` transaction would fail. Since, we should be able to determine if this race condition was triggered, we can differentiate between a race condition failure and a failure for unknown reasons. Thus, we can tell the event watching loop to continue if the `InitializeRound` transaction failed due to a race condition and tell the event watching loop to stop if the `InitializeRound` transaction failed for unknown reasons as a safeguard to prevent the user from losing too much money to transaction costs. If the event watching loop continues, this function will be called again when the next block is mined.

Reply:
Although I'm not sure why the check for whether the current round is initialized uses an explicit comparison between `LastInitializedRound` and `CurrentRound` as opposed to just calling `CurrentRoundInitialized`...I think we should just be able to replace the explicit comparison with a single call to `CurrentRoundInitialized`...

Reply:
Ahh sorry I missed that part.  Yeah I think we can just use `CurrentRoundInitailized` to do the check, and then do the check for the error to catch the condition when someone initialized the round in the same block that we made this call.